File: db4on/core/src/core/com/db4o/JDK.cs
namespace com.db4o
{
	/// <exclude></exclude>
	public class JDK
	{
		internal virtual j4o.lang.Thread addShutdownHook(j4o.lang.Runnable a_runnable)
		{
			return null;
		}

		internal virtual com.db4o.types.Db4oCollections collections(com.db4o.YapStream a_stream
			)
		{
			return null;
		}

		internal virtual j4o.lang.Class constructorClass()
		{
			return null;
		}

		internal virtual object createReferenceQueue()
		{
			return null;
		}

		public virtual object createWeakReference(object obj)
		{
			return obj;
		}

		internal virtual object createYapRef(object a_queue, com.db4o.YapObject a_yapObject
			, object a_object)
		{
			return null;
		}

		internal virtual object deserialize(byte[] bytes)
		{
			throw new com.db4o.ext.Db4oException(com.db4o.Messages.NOT_IMPLEMENTED);
		}

		internal virtual void forEachCollectionElement(object a_object, com.db4o.foundation.Visitor4
			 a_visitor)
		{
		}

		internal virtual string format(j4o.util.Date date, bool showTime)
		{
			return date.ToString();
		}

		internal virtual object getContextClassLoader()
		{
			return null;
		}

		internal virtual object getYapRefObject(object a_object)
		{
			return null;
		}

		internal virtual bool isCollectionTranslator(com.db4o.Config4Class a_config)
		{
			return false;
		}

		public virtual int ver()
		{
			return 1;
		}

		internal virtual void killYapRef(object obj)
		{
		}

		internal virtual void lockFile(object file)
		{
			lock (this)
			{
			}
		}

		/// <summary>
		/// use for system classes only, since not ClassLoader
		/// or Reflector-aware
		/// </summary>
		internal virtual bool methodIsAvailable(string className, string methodName, j4o.lang.Class[]
			 _params)
		{
			return false;
		}

		internal virtual void pollReferenceQueue(com.db4o.YapStream a_stream, object a_referenceQueue
			)
		{
		}

		public virtual void registerCollections(com.db4o.reflect.generic.GenericReflector
			 reflector)
		{
		}

		internal virtual void removeShutdownHook(j4o.lang.Thread a_thread)
		{
		}

		public virtual j4o.lang.reflect.Constructor serializableConstructor(j4o.lang.Class
			 clazz)
		{
			return null;
		}

		internal virtual byte[] serialize(object obj)
		{
			throw new com.db4o.ext.Db4oException(com.db4o.Messages.NOT_IMPLEMENTED);
		}

		internal virtual void setAccessible(object a_accessible)
		{
		}

		internal virtual bool isEnum(com.db4o.reflect.Reflector reflector, com.db4o.reflect.ReflectClass
			 clazz)
		{
			return false;
		}

		internal virtual void unlockFile(object file)
		{
			lock (this)
			{
			}
		}

		public virtual object weakReferenceTarget(object weakRef)
		{
			return weakRef;
		}

		public virtual com.db4o.reflect.Reflector createReflector(object classLoader)
		{
			return null;
		}
	}
}

File: db4oj/core/src/com/db4o/internal/JDK.java
/* Copyright (C) 2004 - 2007   db4objects Inc.   http://www.db4o.com */

package com.db4o.internal;

import java.lang.reflect.*;
import java.net.*;
import java.util.*;

import com.db4o.config.*;
import com.db4o.ext.*;
import com.db4o.foundation.*;
import com.db4o.internal.handlers.*;
import com.db4o.reflect.*;
import com.db4o.reflect.generic.*;
import com.db4o.types.*;

/**
 * @exclude
 * @sharpen.ignore
 */
public class JDK {
	
    /** @param runnable */
	Thread addShutdownHook(Runnable runnable){
		return null;
	}
	
	/**
	 * @param transaction
	 * @deprecated
	 **/
	Db4oCollections collections(Transaction transaction){
	    return null;
	}
    
    Class constructorClass(){
        return null;
    }
	
	Object createReferenceQueue() {
		return null;
	}

    public Object createWeakReference(Object obj){
        return obj;
    }
    
    /**
     * @param queue
     * @param ref
     * @param obj
     */
	Object createActivateObjectReference(Object queue, ObjectReference ref, Object obj) {
		return null;
	}
	
	/** @param bytes */
	Object deserialize(byte[] bytes) {
    	throw new Db4oException(Messages.NOT_IMPLEMENTED);
    }

	/**
	 * @param clazz
	 * @param config
	 */
    public Config4Class extendConfiguration(ReflectClass clazz, Configuration config, Config4Class classConfig) {
    	return classConfig;
    }

    void forEachCollectionElement(Object obj, Visitor4 visitor) {
        Enumeration e = null;
        if (obj instanceof Hashtable) {
            e = ((Hashtable)obj).elements();
        } else if (obj instanceof Vector) {
            e = ((Vector)obj).elements();
        }
        if (e != null) {
            while (e.hasMoreElements()) {
                visitor.visit(e.nextElement());
            }
        }
	}
	
    /** @param showTime */
    String format(Date date, boolean showTime) {
		return date.toString();
	}
	
	Object getContextClassLoader(){
		return null;
	}

	/** @param obj */
	Object getYapRefObject(Object obj) {
		return null;
	}
    
    boolean isCollectionTranslator(Config4Class config) {
        if (config != null) {
            ObjectTranslator ot = config.getTranslator();
            if (ot != null) {
                return ot instanceof THashtable;
            }
        }
        return false;
    }
    
   public boolean isConnected(Socket socket){
       return socket != null;
   }

	public int ver(){
	    return 1;
	}
	
	/** @param obj */
	void killYapRef(Object obj){
		
	}
	
	/**
	 * @param classLoader
	 */
	public Class loadClass(String className, Object classLoader) throws ClassNotFoundException {
	    // We can't use the ClassLoader here since JDK get's converted to .NET
	    // Functionality is overridden in JDKReflect 
		return Class.forName(className);
	}
	
	/**
	 * @param path
	 * @param file
	 */
	synchronized void lockFile(String path,Object file){
	}
	
	/**
     * use for system classes only, since not ClassLoader
     * or Reflector-aware
	 * @param className
	 * @param methodName
	 * @param params
	 */
	boolean methodIsAvailable(String className, String methodName, Class[] params) {
    	return false;
    }

	public long nanoTime() {
		throw new NotImplementedException();
	}
	
	/**
	 * @param session
	 * @param referenceQueue
	 */
	void pollReferenceQueue(ObjectContainerBase session, Object referenceQueue) {
		
	}
	
    /** @param reflector */
	public void registerCollections(GenericReflector reflector) {
		
	}
	
    /** @param thread */
	void removeShutdownHook(Thread thread){
		
	}
	
    /** @param clazz */
	public Constructor serializableConstructor(Class clazz){
	    return null;
	}
	
    /** @param obj */
	byte[] serialize(Object obj) throws Exception{
    	throw new Db4oException(Messages.NOT_IMPLEMENTED);
    }

    /** @param accessibleObject */
	void setAccessible(Object accessibleObject) {
	}
    
	/**
	 * @param reflector
	 * @param clazz
	 */
    boolean isEnum(Reflector reflector, ReflectClass clazz) {
        return false;
    }
	
	/**
	 * @param path
	 * @param file
	 */
    synchronized void unlockFile(String path,Object file) {
	}
    
    public Object weakReferenceTarget(Object weakRef){
        return weakRef;
    }
    
    /** @param classLoader */
    public Reflector createReflector(Object classLoader) {
    	return null;
    }

	/** @param clazz */
    public Reflector reflectorForType(Class clazz) {
		return null;
	}

    /** @param container */
    public NetTypeHandler[] types(ObjectContainerBase container) {
        return new NetTypeHandler[]{};
    }

	public boolean useNativeSerialization() {
		return true;
	}
}
